tests/run_all_tests.py

```
@@ -0,0 +1,242 @@
+            "network_connectivity": [],
+            "scraper_functionality": [],
+            "dashboard_ui": [],
+            "environment_setup": []
+        }
+        
+        recommendations = []
+        
+        # Check API tests
+        api_result = self.test_results.get("API Endpoints")
+        if api_result and not api_result["success"]:
+            if "connection" in api_result["stderr"].lower() or "network" in api_result["stderr"].lower():
```

Ruff (0.11.9)

Use a single `if` statement instead of nested `if` statements

(SIM102)